FileManager.php

```diff
@@ -368,6 +370,32 @@ private function unsafeDeduplicate(string $filename): string
         return $candidate;
     }
 
+    /**
+     * Use finder to extract files list from a directory
+     * @param $uri is the directory uri
+     * @param $pattern is an optionnal patteren restriction (like '*.csv')
+     * @param $createDirectoryIfNotExists is False by default, if true the $uri
+     *        directory would be created if not exists
+     */
+    public function ls(string $uri, $pattern='*', $createDirectoryIfNotExists=False): Finder
```

Comment:
`$createDirectoryIfNotExists = false` this is not Python :)

Reply:
`$pattern = '*'`

Reply:
done